File: erlang/leap/src/leap.erl
-module(leap).

-export([leap_year/1, test_version/0]).

-spec leap_year(integer()) -> boolean().
leap_year(Year) when Year rem 4 == 0, Year rem 100 /= 0; Year rem 400 == 0 ->
    true;
leap_year(_Year) ->
    false.

-spec test_version() -> integer().
test_version() ->
    4.

File: python/leap/leap.py
def is_leap_year(year: int) -> bool:
    """
    Determine if the given year is a leap year.

    year -- An integer year.
    returns -- The leap year status as true/false.
    """

    return year % 400 == 0 or (year % 100 != 0 and year % 4 == 0)
